horovod/torch/adapter.cc

```diff
@@ -146,7 +146,7 @@ void ThrowIfError(Status status) {
     throw std::runtime_error(status.reason());
   case StatusType::INVALID_ARGUMENT:
     throw std::invalid_argument(status.reason());
-  default: // Includes UNKNOWN_ERROR
+  default: // Includes UNKNOWN_ERROR and OUT_OF_RANGE_ERROR
```

Comment:
Let's just have an explicit case, like we do for ABORTED error.

Reply:
changed.